### ISSUE TITLE
TinyMCE issue in forums

### ISSUE THREAD

**Issue by adira-g** [SocialEngine/phpv4-issues#719]
(Thanks for reporting an issue! Please make sure you click the link above to view the issue guidelines, then fill out the blanks below.)

What are the steps to reproduce this issue?
-------------------------------------------
1. open a new topic in forums
2. use tinyMCE and post something containing a numbered list
3. edit html code and click okay 
OR just publish the post and edit again (same result)

What happens?
-------------
between each line of the list, there are <o></o> tags added, consequently if you save, it won't count anymore, but each line begins with 1. Also extra breaks are added by saving after editing the html code

What were you expecting to happen?
----------------------------------
html code shouldn't change, just by closing the box or saving without changes
html code shouldn't change, when I edit the post

Any logs, error output, etc?
----------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.) 

Any other comments?
-------------------

by the way: On blog posts, tinyMCE works fine

see attached pics:

![screenshot 2017-04-29 17 48 14](https://cloud.githubusercontent.com/assets/27999622/25556917/bfb57a94-2d06-11e7-9676-e092d6aa8c51.jpg)

editing html before saving:
![screenshot 2017-04-29 17 56 35](https://cloud.githubusercontent.com/assets/27999622/25556927/eb324c56-2d06-11e7-9094-f89a9a9b962d.jpg)

after closing the source code box without any changes:
![screenshot 2017-04-29 17 48 52](https://cloud.githubusercontent.com/assets/27999622/25556934/288dc3be-2d07-11e7-8294-0cca3a2ec77d.jpg)

again the source code:
![screenshot 2017-04-29 18 11 11](https://cloud.githubusercontent.com/assets/27999622/25556941/62ee3a66-2d07-11e7-84f6-25bf2e98897f.jpg)

and once more, clicking ok and open source code again looks like this:
![screenshot 2017-04-29 18 11 31](https://cloud.githubusercontent.com/assets/27999622/25556951/81f123b0-2d07-11e7-90e0-3af290ab4050.jpg)

What versions of software are you using?
----------------------------------------
**Operating System:** …

**SocialEngine PHP Version:** 4.9.0

**Comment by Mthorganby** [comment]
Note the "bullet list" has the same problem, the dots are not showing at the beginning of the line

Only when you use hold ALT + 7 and release a dot is showing
<span style="margin:30px;">• abcdefghijklim </span>
<br />
<span style="margin:0px;">• abcdefghijklim </span>

The <br /> is deleted and replace by an empty line, which is resulting Chrome only adding the <span> behind each other to form one line of text (with margins).


If you save the bullet list and look in Chrome the <ol> is converted to <ul>

In the theme.css : 

```
ul
{
  list-style-type: none;
}
```

Where are my nice little bullets gone??

**Comment by adira-g** [comment]
How does it work for you in blog entries? Using the same installation, there are no issues with the editor in blogs. What is the difference? In each case, the tinyMCE appears, but the behavior is different.

Example of blog entries:
![screenshot 2017-04-29 18 55 28](https://cloud.githubusercontent.com/assets/27999622/25557237/f775132a-2d0d-11e7-88a1-b748bd8cd90c.jpg)
![screenshot 2017-04-29 18 55 34](https://cloud.githubusercontent.com/assets/27999622/25557238/ff79f5ae-2d0d-11e7-8144-645b556798be.jpg)

looks published as expected:
![screenshot 2017-04-29 19 00 42](https://cloud.githubusercontent.com/assets/27999622/25557250/358f99e6-2d0e-11e7-9c80-488d00d4cef0.jpg)

and behind the scene html output:
![screenshot 2017-04-29 19 01 51](https://cloud.githubusercontent.com/assets/27999622/25557267/93fa9cc4-2d0e-11e7-918b-45b3ee94e715.jpg)

**Comment by Mthorganby** [comment]
You are in Blogs which is working, we are in Forums, where is goes wrong, from the out-of -the-box install aka fresh install

**Comment by Mthorganby** [comment]
@DonnaScriptTechs  this is a BUG... i will be posting more :)

**Comment by DonnaScriptTechs** [comment]
No need to tag. :) We see the reports. 

https://github.com/SocialEngine/phpv4-issues/issues/669 looks to be related to that bug.

**Comment by adira-g** [comment]
@Mthorganby 

> You are in Blogs which is working, we are in Forums, where is goes wrong, from the out-of -the-box install aka fresh install

Yes I know, it's me, who opened this issue ;-)
I just wanted to point out, that it works on blogs. 

@DonnaScriptTechs 
Yes, it seems to be somehow related. I hope my report helps to fix this bug quickly. It's probably not a big thing.

**Comment by Elshara** [comment]
Can reproduce here as well.

**Comment by SpiritWolfie** [comment]
Can confirm, this is a MAJOR issue not only does this affect our ability's to space posts and format posts

it's also taking links to other forum posts (if I create Thread 1 and post a link asking people to see thread B) it's redirecting to /home/ even if the URL is correct....

This needs to be fixed ASAP Please, this is affecting the ability to use our text editor on a global scale at this point. Making forums look cluttered etc....

**Comment by Elshara** [comment]
I'm about ready to just toss the forums plugin goodbye and just use
blogs and categories at this point.

**Comment by DonnaScriptTechs** [comment]
If you set it to use bbcode instead, for now, does that work?

**Comment by Elshara** [comment]
I did enable it as well as selective html tags. I usually just
override page source.

**Comment by Mthorganby** [comment]
Hi @DonnaScriptTechs 
It is set to Yes as standard to use the BBcode, still doesn't work

Allowed HTML: 
blockquote, strong, b, em, i, u, strike, sub, sup, p, div, pre, address, h1, h2, h3, h4, h5, h6, span, ol, li, ul, a, img, embed, br, hr, iframe

IF you set use BBcode to NO then:
The actual html looks like after saving:

```
<p>begin test</p>
<ul>
<li>lkjlkj</li>
<li>lkjljk</li>
<li>lkjljl</li>
<li>lkjljl</li>
</ul>
<p>end test</p>
```


IF you set to use BBcode to YES then
The actual html looks like after saving:
```

begin test





<ul>
<ul>
<li>lkjlkj</li>
</ul>
</ul>
 

<ul>
<ul>
<li>lkjljk</li>
</ul>
</ul>
 

<ul>
<ul>
<li>lkjljl</li>
</ul>
</ul>
 

<ul>
<ul>
<li>lkjljl</li>
</ul>
</ul>




end test
```

Somewhere where is produces the code to save it goes wrong

**Comment by DonnaScriptTechs** [comment]
I've added this as a separate issue for us to fix so we can double check it in case the announcement fix doesn't fix this issue too.

**Comment by adira-g** [comment]
I want to add another strange behavior in the forums (SE 4.9.0), maybe it is even another issue. But let's see, if anyone can re-produce this issue:

If you click on "quote" in a forum post, it opens the editor and copies the text as a quote into the editor. But it wraps the text into [blockquote] tags, which is not a valid bb code, is it? It appears then (as expected) not as quote, one has to manually change the the tags from bb-code-tags to html tags [blockquote] into <blockquote> or change the bb-code from [blockquote] to [quote].

**Comment by DonnaScriptTechs** [comment]
Since both tinymce and bbcode are broken right now it is hard for me to test but I'll try. This is being worked on for the tiny and bbcode issue originally reported.

**Comment by Mthorganby** [comment]
@adira-g we are having issues on SE 4.9.1 not SE 4.9.0, upgrade and see if you still have this issue. You can add  [quote] to the html allowed in Member Level Settings.
Standard allowed is: blockquote, strong, b, em, i, u, strike, sub, sup, p, div, pre, address, h1, h2, h3, h4, h5, h6, span, ol, li, ul, a, img, embed, br, hr, iframe

**Comment by adira-g** [comment]
Excellent, Donna, thank you :-)

**Comment by DonnaScriptTechs** [comment]
I tested on 4.8.13 (just to see) and it uses blockquote too but it works there. I can add this to the issue report in case it should be changed.

**Comment by adira-g** [comment]
Yes Donna, many users used it on our site and it worked on 4.8.13; I noticed the issue because on of our members was pointing me on it ;-)

**Comment by SpiritWolfie** [comment]
Can confirm this problem on my end as well, it's part of the statement I wrote as well, this was something I saw as well.

**Comment by DonnaScriptTechs** [comment]
This is fixed for 4.9.2 when it is released. If you still have this issue after we release, please open a new bug report.
